src/command.c

```diff
@@ -564,7 +564,19 @@ static bool cmd_rtt(target_s *t, int argc, const char **argv)
 			if (rtt_ident[i] == '_')
 				rtt_ident[i] = ' ';
 		}
-	} else if (argc == 2 && strncmp(argv[1], "ram", command_len) == 0)
+	} else if (argc == 3 && strncmp(argv[1], "down", command_len) == 0) {
+		/*
+		 * here we bypass the if statement to save flash
+		 * as it's not critical to check if the rtt block is found
+		 */
+		if(rtt_found || true) {
```

Comment:
Either include the check, or entirely eliminate it please. Likewise, proper capitalisation - "Here" in the above comment, "The" in the gdb_out() below. Please drop the braces on the else, they're unnecessary.

Reply:
If I include the check, the error message adds to the .text segment, so I just print something generic like "error"? I think it is good to have this in place just in case and also it is OK to bypass, bacause it is not at all critical. So if someone is browsing the code can re-enable it.

Reply:
Given what the message is trying to express, perhaps `gdb_print("rtt not found\n");`? It expresses the same information without using quite so much Flash - we'd really prefer this and the check to be enabled, or just entirely eliminate the check. The only reason the `|| true` is not causing additional Flash usage is because of the Dead-Code Elimination pass and we do not want to be relying on that, it's implementation-defined behaviour.